### PR TITLE
Added password dialog

### PR DESCRIPTION
**Pull request by Mykhailo96**

### CODE REVIEW

**Comment by alexchmykhalo** [comment]
Please, check documentation for QDialog::accept() and QDialog::reject() and as well as QDialogButtonBox class. It is not only provide native layout for buttons (on different OSs) but also allows to assign button role, so we do not need to create any custom slots.

Please, make changes accordingly to remarks.

**Comment by alexchmykhalo** [comment]
* Error dialog has no method for changing long error description
* Error dialog has no method for changiing short error description.
* No design from mockup applied

File: src/app/ui/passworddialog.cpp [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+}
+
+PasswordDialog::~PasswordDialog()
+{
+    delete ui;
+}
+
+void PasswordDialog::accept()
+{
+    PasswordDialog::~PasswordDialog();
```
Comment (Houd1ny) [inline src/app/ui/passworddialog.cpp]: u don`t need to remove password dialog
You should read documentation of QDialog
" This includes deleting the dialog while it is visible."
You should in ui connect ok clicked to QDialog accepted.

File: src/app/ui/passworddialog.h [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+}
+
+class PasswordDialog : public QDialog
+{
+    Q_OBJECT
+
+public:
+    explicit PasswordDialog(QWidget *parent = 0);
+    ~PasswordDialog();
+
```
Comment (Houd1ny) [inline src/app/ui/passworddialog.h]: i canse dialog accepted
how can i get data from dialog?
getters should be

**Comment by galynabutovych** [comment]
- what is the point of connecting to slot that returns a value?
  QString PasswordDialog::username()
- use QString username() const; as a normal public function, not as a slot

**Comment by galynabutovych** [comment]
Please, set QLineEdit::Password for password line edit using setEchoMode and I will merge your pull request
git pull before making changes

**Pull request merged**